### ISSUE TITLE
yotta target <targetname> should accept version specs

### ISSUE THREAD

**Issue by autopulated** [ARMmbed/yotta#113]
Targets should also be subject to semantic versioning, which means it must be possible to install particular versions of targets.

(possible by design, but the command line parsing needs fixing)

**Comment by jaustin** [comment]
Furthermore, we seem to 'cache' targets quite aggressively and need to rm -rf the yotta_targets/<target> dir if we want to fetch the newest version of a target. yt up doesn't seem to do this.

EG from a user over email: "this didn't work for me - `rm -rf yotta_targets/bbc-microbit-classic-gcc/`  seemed to work though."

**Comment by autopulated** [comment]
`yt up` _does_ check for updated target versions (from the registry). I'm not aware of any way of making it not do that, if you can reproduce a situation where it doesn't, please open a separate issue.

**Comment by msprotz** [comment]
What I'm observing is that a binary compiled with yotta build on my machine has the bootloader, while doing the same thing through the compile service gives a binary that does not have the bootloader, it seems.

I'll do more tests and investigate, hopefully I'll have some more steps to reproduce the issue.

**Comment by jaustin** [comment]
@msprotz your answer is specific to the target you're using and the compile service, not yotta itself necessarily - though I will try to reproduce the cases where the targets have not updated automatically, because it might be the compile service is hitting them...

**Comment by msprotz** [comment]
Right. I just checked and the binaries generated through the compile service now have the bootloader with them, so it seems like we're good. There may have been some cache in the middle that expired since yesterday.

**Comment by rainierwolfcastle** [comment]
http://jira-dev0.cambridge.arm.com/browse/IOTSFW-906
